pkg/client/client.go

```diff
@@ -139,6 +158,94 @@ func (c *tlsBootstrapClientImpl) GetBootstrapToken(ctx context.Context) (string,
 	return string(execCredentialBytes), nil
 }
 
+func (c *tlsBootstrapClientImpl) isKubeConfigStillValid() (bool, error) {
+	c.logger.Debug("checking if kubeconfig exists...")
+
+	_, err := os.Stat(c.kubeConfigPath)
+	if os.IsNotExist(err) {
+		c.logger.Debug("kubeconfig does not exist. bootstrapping will continue")
+		return false, nil
+	}
+	if err != nil {
+		return false, fmt.Errorf("error reading existing bootstrap kubeconfig %s: %v", c.kubeConfigPath, err)
+	}
+
+	isValid, err := isClientConfigStillValid(c.kubeConfigPath)
+	if err != nil {
+		return false, fmt.Errorf("unable to load kubeconfig: %v", err)
+	}
+	if isValid {
+		c.logger.Debug("kubeconfig is valid. bootstrapping will not continue")
+		return true, nil
+	}
+
+	c.logger.Debug("kubeconfig is invalid. bootstrapping will continue")
+	return false, nil
+}
+
+// copied from https://github.com/kubernetes/kubernetes/blob/e45f5b089f770b1c8a1583f2792176bfe450bb47/pkg/kubelet/certificate/bootstrap/bootstrap.go#L231
+// isClientConfigStillValid checks the provided kubeconfig to see if it has a valid
+// client certificate. It returns true if the kubeconfig is valid, or an error if bootstrapping
+// should stop immediately.
+func isClientConfigStillValid(kubeconfigPath string) (bool, error) {
+	_, err := os.Stat(kubeconfigPath)
+	if os.IsNotExist(err) {
+		return false, nil
+	}
+	if err != nil {
+		return false, fmt.Errorf("error reading existing bootstrap kubeconfig %s: %v", kubeconfigPath, err)
```

Comment:
probably better to not fail bootstrapping if we can't read the existing kubeconfig, I'd rather lean towards just continuing the bootstrap process